### DISCUSSION TITLE
Class components in Supabase

### DISCUSSION THREAD

**Discussion by jessedoka** [supabase/supabase#3604] [Questions] [answered]
Hey for my project i have written my compoenent with a class class component, I am not sure how to use supabase with class compoenents. Any Help?

**Comment by trankhacvy** [comment, marked as answer]
@chukkyiii I assume you mention about Reactjs class-component. I think you can use supabase like the way you call ajax or API, just put the code in `componentDidMount` function
```

async componentDidMount() {
    this.setState({ isLoading: true });

    try {
      const { error, data } = await supabase.from('users').select("*")
      if(error) {
            throw error;
      }
      this.setState({
        users: data,
        isLoading: false
      });
    } catch (error) {
      this.setState({
        error,
        isLoading: false
      });
    }
  }
 
```

**Comment by jessedoka** [comment]
Thanks for this by the way!
It works!
